bundles/framework/featuredata/plugin/MapSelectionPlugin.js

```diff
@@ -0,0 +1,146 @@
+            this.clearDrawing();
+            // disable all draw controls
+            this._toggleControl(null, keepDrawMode);
+        },
+        setSelectFromAllLayers: function (selectAll) {
+            this.selectFromAllLayers = selectAll;
+        },
+        isSelectFromAllLayers: function () {
+            return this.selectFromAllLayers;
+        },
+        /**
+         * @method _toggleControl
+         * Enables the given draw control
+         * Disables all the other draw controls
+         * @param {String} drawMode draw control to activate (if undefined, disables all
+         * controls)
+         * @private
+         */
+        _toggleControl: function (drawMode, keepDrawMode = false) {
+            let key;
+            let control;
+            if (keepDrawMode && !drawMode) {
+                // keep previous draw mode active
+            } else {
+                this.currentDrawMode = drawMode;
+            }
+            for (key in this.drawControls) {
+                if (this.drawControls.hasOwnProperty(key)) {
+                    control = this.drawControls[key];
+                    if (this.currentDrawMode === key) {
+                        control();
+                    }
+                }
+            }
+        },
+
+        /**
+         * Initializes the plugin:
+         * - drawControls
+         * - registers for listening to requests
+         * @param sandbox reference to Oskari sandbox
+         * @method
+         */
+        _initImpl: function () {
+            const me = this;
+            const sb = this.getSandbox();
+
+            me.drawControls = {
+                point: function () {
+                    sb.postRequestByName('DrawTools.StartDrawingRequest', [
+                        me.DRAW_REQUEST_ID,
+                        'Point'
+                    ]);
+                },
+                line: function () {
+                    sb.postRequestByName('DrawTools.StartDrawingRequest', [
+                        me.DRAW_REQUEST_ID,
+                        'LineString'
+                    ]);
+                },
+                polygon: function () {
+                    sb.postRequestByName('DrawTools.StartDrawingRequest', [
+                        me.DRAW_REQUEST_ID,
+                        'Polygon'
+                    ]);
+                },
+                square: function () {
+                    sb.postRequestByName('DrawTools.StartDrawingRequest', [
+                        me.DRAW_REQUEST_ID,
+                        'Square'
+                    ]);
+                },
+                circle: function () {
+                    sb.postRequestByName('DrawTools.StartDrawingRequest', [
+                        me.DRAW_REQUEST_ID,
+                        'Circle'
+                    ]);
+                }
+            };
+        }
+    }, {
+        extend: ['Oskari.mapping.mapmodule.plugin.AbstractMapModulePlugin'],
```

Comment:
This doesn't really reference the map in any way so we could implement this as a regular file/part of the "PopupHandler" instead. This used to call the map directly to activate drawing tools etc, but now it just uses the requests without needing direct reference to the mapmodule.